include/Zycore/Types.h

```diff
@@ -333,6 +333,84 @@ typedef char* ZyanCharPointer;
  */
 typedef const char* ZyanConstCharPointer;
 
+/* ============================================================================================== */
+/* Type conversion utils                                                                          */
+/* ============================================================================================== */
+
+/**
+ * Collection of functions for swapping byte order in 16/32/64-bit values respectively.
+ *
+ * @param   x       16/32/64-bit value.
+ *
+ * @return  16/32/64-bit value with byte order swapped.
+ */
+#if defined(ZYAN_MSVC)
+#   if defined(ZYAN_NO_LIBC)
+#ifdef __cplusplus
+extern "C" {
+#endif
+    unsigned short __cdecl _byteswap_ushort(unsigned short);
+    unsigned long  __cdecl _byteswap_ulong(unsigned long);
+    unsigned __int64 __cdecl _byteswap_uint64(unsigned __int64);
+#ifdef __cplusplus
+}
+#endif
+#   elif defined(ZYAN_KERNEL)
+#       include <wdm.h>
+#   else
+#       include <stdlib.h>
+#   endif
+
+#   define ZYAN_BYTESWAP16(x) _byteswap_ushort((x))
+#   define ZYAN_BYTESWAP32(x) _byteswap_ulong((x))
+#   define ZYAN_BYTESWAP64(x) _byteswap_uint64((x))
+#elif ZYAN_HAS_BUILTIN(__builtin_bswap64)
```

Comment:
Let's check presence of all the `__builtin` variants here. It's possible to disable builtins individually with some compiler flag that I forgot. I had to occasionally use this in build systems to work around bugs in upstream libraries (`glibc` specifically).